### PR TITLE
Working Ten Pin Bowling in JavaScript with basic HTML display for reference.

### PR DESCRIPTION
**Pull request by RichardCharman**

### CODE REVIEW

File: bowling.html [hunk 1]
```diff
@@ -0,0 +1,49 @@
+<!DOCTYPE html>
+<html>
+  <head>
+    <meta charset="utf-8">
+    <title>Ten Pin Bowling</title>
+    <script type="text/javascript" src="src/Bowling.js"></script>
+  </head>
+  <body>
```
Comment (danldb) [inline bowling.html]: why inline styles? Use a separate stylesheet.

File: bowling.html [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+      <div style='border-bottom: 1px outset blue; padding-bottom: 10px'>
+        <class id="bonusRoll1"></class> <class id="bonusRoll2"></class>
+      </div>
+      <div style='font-size:30px; padding: 10px'>
+        Your Total Score is <class id="score"></class>
+      </div>
+      <div style='padding: 10px'>
+        <button type="button" onclick="reloadPage()" > NEW GAME </button>
+      </div>
+      <script>
```
Comment (danldb) [inline bowling.html]: again, use external script files rather than inline scripts.

File: index.php [hunk 3]
```diff
@@ -0,0 +1 @@
+<?php include_once("bowling.html"); ?>
```
Comment (danldb) [inline index.php]: nice solution

File: bowling.html (hunk 1 above)
Comment (RichardCharman) [inline bowling.html]: Didn't finish, would've gone there with a little more time.

File: lib/jasmine-2.3.4/boot.js [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+
+  /**
+   * Helper function for readability above.
+   */
+  function extend(destination, source) {
+    for (var property in source) destination[property] = source[property];
+    return destination;
+  }
+
+}());
```
Comment (danldb) [inline lib/jasmine-2.3.4/boot.js]: never commit your vendor code (i.e. any code that you did not write yourself). Better to use bower or npm to install the code, or alternatively use a CDN. You coudl also consider putting all your vendor code in a folder called 'vendor' and then putting that folder in .gitignore.

File: spec/BowlingSpec.js [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+
+    it('knows if player has rolled a gutter ball', function() {
+      game.hitPin(2);
+      game.hitPin(4);
+      game.hitPin(7);
+      expect(game.frameScore).toEqual(3);
+      game.hitPin(0);
+      expect(game.frameScore).toEqual(3);
+    });
+
```
Comment (danldb) [inline spec/BowlingSpec.js]: Write some function to remove repetition.

File: src/Bowling.js [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+    this.rollBall();
+    this.bonusPoints();
+    this.strikeBonusA-=1;
+    this.strikeBonusB-=1;
+    this.spareBonusA-=1;
+    this.spareBonusB-=1;
+    this.scoreCard.push(this.knockedDown);
+    this.frame += 1;
+    if(this.frame==10){
+    this.eleventhFrame();
```
Comment (danldb) [inline src/Bowling.js]: indentation

Comment (danldb) [inline src/Bowling.js]: So many variables is a code smell - consider ways to extract this into functions that calculate 'on-the-fly' the current state of the object.

File: src/Bowling.js [hunk 7]
```diff
@@ -0,0 +1,121 @@
...
+       if(this.spareBonusA==1) {
+        this.spareBonusB=2;
+      } else {
+        this.spareBonusA=2;
+      }
+     }
+     return this.secondRoll
+   };
+};
+
```
Comment (danldb) [inline src/Bowling.js]: Beware the pyramid of doom. Multiple if else statements are a sign that you should extract your logic into an object.

File: src/Bowling.js [hunk 8]
```diff
@@ -0,0 +1,121 @@
...
+  }
+  if(this.spareBonusA==1) {
+    this.spareBonusA=0;
+    return this.score += this.firstRoll
+  };
+  if(this.spareBonusB==1) {
+    this.spareBonusB=0;
+    return this.score += this.firstRoll
+  };
+ };
```
Comment (danldb) [inline src/Bowling.js]: multiple ifs

**Comment by danldb** [comment]
:checkered_flag: 

Good attempt Rich. Focus more on SRP, and keeping your logic in control. Although I have awarded a pass for this, I would advise that you should refactor this at the end of the course, based on comments.

**Pull request closed without merging**